pkg/environments/gitops.go

```diff
@@ -601,3 +608,95 @@ func CreateNestedRequirementDir(dir string, requirementName string, requirementD
 	}
 	return nil
 }
+
+// AddAppMetaData applies chart metadata to an App resource
+func AddAppMetaData(chartDir string, app *jenkinsv1.App, repository string) (*jenkinsv1.App, error) {
+	metadata, err := helm.LoadChartFile(filepath.Join(chartDir, "Chart.yaml"))
+	if err != nil {
+		return nil, errors.Wrapf(err, "error loading chart from %s", chartDir)
+	}
+	if app.Annotations == nil {
+		app.Annotations = make(map[string]string)
+	}
+	app.Annotations[helm.AnnotationAppDescription] = metadata.GetDescription()
+	repoURL, err := url.Parse(repository)
+	if err != nil {
+		return nil, errors.Wrap(err, "Invalid repository url")
+	}
+	app.Annotations[helm.AnnotationAppRepository] = util.StripCredentialsFromURL(repoURL)
+	if app.Labels == nil {
+		app.Labels = make(map[string]string)
+	}
+	app.Labels[helm.LabelAppName] = metadata.Name
+	app.Labels[helm.LabelAppVersion] = metadata.Version
+	return app, nil
+}
+
+// UpdateAppResource updates app resource template with app specific metadata
+func UpdateAppResource(helmer helm.Helmer, fetchedChartDir string, outputDir string, name string, repository string) error {
+	outputTemplateDir := filepath.Join(outputDir, "templates")
+	templatesDirExists, err := util.DirExists(outputTemplateDir)
+	if err != nil {
+		return err
+	}
+	if !templatesDirExists {
+		os.Mkdir(outputTemplateDir, os.ModePerm)
+	}
+	helmClient, foundHelmCLI := helmer.(*helm.HelmCLI)
```

Comment:
This seems really wrong as Helmer is supposed to hide these details. I would suggest modifying `Helmer` to add the function you need.